bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/TextLayout.java

```diff
@@ -757,7 +759,29 @@ void drawInPixels (GC gc, int x, int y, int selectionStart, int selectionEnd, Co
  */
 public void draw (GC gc, int x, int y, int selectionStart, int selectionEnd, Color selectionForeground, Color selectionBackground, int flags) {
 	checkLayout();
-	drawInPixels(gc, DPIUtil.autoScaleUp(getDevice(), x), DPIUtil.autoScaleUp(getDevice(), y), selectionStart, selectionEnd, selectionForeground, selectionBackground, flags);
+	nativeZoom = gc.data.nativeDeviceZoom;
+	drawInPixels(gc, DPIUtil.autoScaleUp(getDevice(), x, getZoom(gc)), DPIUtil.autoScaleUp(getDevice(), y, getZoom(gc)), selectionStart, selectionEnd, selectionForeground, selectionBackground, flags);
+}
+
+private int getNativeZoom(GC gc) {
+	if (gc != null) {
+		return gc.data.nativeDeviceZoom;
+	} else if(font != null) {
+		return font.zoom;
+	}
+	return DPIUtil.getNativeDeviceZoom();
+}
```

Comment:
This looks like some incomprehensible catch-all logic to me. It simply tries to retrieve the zoom from whatever is available and otherwise falls back to a default value.
1. I do not understand why the default value makes any sense. Won't that be invalid in a multi-monitor scenario?
2. Wouldn't it be better to decide for every calling place which information _must_ be available and use that for retrieving the zoom, so that in case that fails we are sure what value is expected and may more easily find out why it is not present?

Reply:
Have beena addressed and revised to a simpler method now.